backend/producer.py

```diff
@@ -86,16 +83,17 @@ def update_power_profile(self, prediction):
             new_prediction = prediction+offset
             self.prediction = new_prediction.combine_first(self.prediction)
 
-    def create_contract(self, consumer, job):
-        id = random.randint(0, 1000)  # TODO: create cool id
+    def create_contract(self, job):
+        current_time = self.manager.clock.now
+        id = self.id + ";" + job.id + ";" + str(current_time)
```

Comment:
id format up for discussion, but this should be unique at least.

Reply:
Looks good, as long as the folks in the front-end can make sense of it